src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3328,14 +3292,85 @@ private IOperation CallNullableMember(IOperation value, SpecialMember nullableMe
 
                 return GetCaptureReference(resultCaptureId, operation);
             }
+
+            void resetConditionalAccessTracker()
+            {
+                Debug.Assert(!_currentConditionalAccessTracker.IsDefault);
+                Debug.Assert(_currentConditionalAccessTracker.Operations.Count == 0);
+                _currentConditionalAccessTracker.Free();
+                _currentConditionalAccessTracker = previousTracker;
+            }
         }
 
         public override IOperation VisitConditionalAccessInstance(IConditionalAccessInstanceOperation operation, int? captureIdForResult)
         {
-            Debug.Assert(_currentConditionalAccessInstance != null);
-            IOperation result = _currentConditionalAccessInstance;
-            _currentConditionalAccessInstance = null;
-            return result;
+            Debug.Assert(!_currentConditionalAccessTracker.IsDefault && _currentConditionalAccessTracker.Operations.Count > 0);
+
+            IOperation testExpression = _currentConditionalAccessTracker.Operations.Pop();
+            return VisitConditionalAccessTestExpression(testExpression);
+        }
+
+        static bool IsConditionalAccessInstancePresentInChildren(IOperation operation)
+        {
+            if (operation is not NoneOperation)
```

Comment:
>if (operation is not NoneOperation) [](start = 12, length = 35)

I think we should have a comment explaining rational behind this condition.

Reply:
If this is just an optimization, I am not sure if it is worth it.

---
In reply to: [573251761](https://github.com/dotnet/roslyn/pull/51032#discussion_r573251761) [](ancestors = 573251761)

Reply:
This was not an optimization. It was, however, a bug. So I've fixed that, and the updates to `ConditionalAccess_13` will reflect this.

---
In reply to: [573401448](https://github.com/dotnet/roslyn/pull/51032#discussion_r573401448) [](ancestors = 573401448,573251761)